hugegraph-api/src/main/java/com/baidu/hugegraph/core/GraphManager.java

```diff
@@ -98,6 +111,114 @@ public void waitGraphsStarted() {
         });
     }
 
+    public HugeGraph cloneGraph(String name, String newName,
+                                String configText) {
+        /*
+         * 0. check and modify params
+         * 1. create graph instance
+         * 2. init backend store
+         * 3. inject graph and traversal source into gremlin server context
+         * 4. inject graph into rest server context
+         */
+        HugeGraph g = graph(name);
+        E.checkArgumentNotNull(g, "The origin graph '%s' doesn't exist", name);
+        E.checkArgumentNotNull(newName, "The graph name can't be null");
+        E.checkArgument(!this.graphs().contains(newName),
+                        "The graph '%s' has existed", newName);
+        PropertiesConfiguration propConfig = this.buildConfig(configText);
+
+        HugeConfig config = (HugeConfig) g.configuration();
+        HugeConfig cloneConfig = (HugeConfig) config.clone();
+        cloneConfig.setDelimiterParsingDisabled(true);
+        // Use the passed config to overwrite the old one
+        propConfig.getKeys().forEachRemaining(key -> {
+            cloneConfig.setProperty(key, propConfig.getProperty(key));
+        });
+        this.checkOptions(cloneConfig);
+        return this.createGraph(cloneConfig);
+    }
+
+    public HugeGraph createGraph(String name, String configText) {
+        E.checkArgumentNotNull(name, "The graph name can't be null");
+        E.checkArgument(!this.graphs().contains(name),
+                        "The graph name '%s' has existed", name);
+
+        PropertiesConfiguration propConfig = this.buildConfig(configText);
+        HugeConfig config = new HugeConfig(propConfig, false);
```

Comment:
Cannot resolve constructor 'HugeConfig(org.apache.commons.configuration.PropertiesConfiguration, boolean);
are you sure? @ @javeme @Linary

Reply:
This constructor was defined in hugegraph-common 1.7.10, it's not yet published to the maven central repository.